posthog/models/filters/mixins/funnel_trends.py

```diff
@@ -0,0 +1,24 @@
+from posthog.constants import FUNNEL_FROM_STEP, FUNNEL_TO_STEP
+from posthog.models.filters.mixins.base import BaseParamMixin
+from posthog.models.filters.mixins.utils import cached_property, include_dict
+
+
+class FunnelTrendsMixin(BaseParamMixin):
+    @cached_property
+    def funnel_from_step(self) -> int:
+        from_step = int(self._data.get(FUNNEL_FROM_STEP, 1))
+        return from_step
+
+    @cached_property
+    def funnel_to_step(self) -> int:
+        to_step = int(self._data.get(FUNNEL_TO_STEP, len(self.entities)))
```

Comment:
looks like this enforces this mixin to go in after entities. Should inherit from it?

Reply:
Well, hm, `FunnelTrendsMixin` can't inherit from `EntitiesMixin` because they both inherit from `BaseParamMixin`, which messes up MRO. So not sure how to _ensure_ that `FunnelTrendsMixin` is used in conjunction with `EntitiesMixin`. Though it is safe, since that's the only way in which it is used.

Reply:
right right. Ugh. 

Makes sense.

Reply:
Oh wait, it doesn't need `BaseParamMixin` if it has `EntitiesMixin`, which should be fine?

Reply:
I tried that but then `Filter`'s MRO is broken. 🤔

Reply:
*resolves conversation* . Hmm, I don't understand MRO enough to model what should happen, without testing it out.

Reply:
I moved responsibility for default values from mixin to the query class, so this isn't a problem anymore :)